feat(clickhouse): add examples section for otlp -> clickhouse (#2699)
flipt-io/flipt: examples/tracing/otlp/README.md

1. Open the Datadog traces UI under the menu item on the left `APM` then `Traces`
1. You should see a list of traces to explore

> ### Clickhouse

Reviewer: on second thought, im thinking this might not really belong here, as if I were looking for examples on how to do analytics with clickhouse, I probably wouldn't look in the `tracing/otlp` directory.

Maybe we can create a new `examples/analytics/clickhouse` directory and move all the clickhouse stuff and this part of this readme there. The fact that it uses OTLP is a bit of implementation detail IMO. WDYT?

Author: @markphelps Yeah good idea, I switched some things around.

#### Introduction

[Clickhouse](https://clickhouse.com/) has alpha support as a destination for traces collected by an open telemetry collector as stated in the [docs](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/clickhouseexporter/README.md). Because of this, the source code for the collector we have to use is [here](https://github.com/open-telemetry/opentelemetry-collector-contrib) as this collector can read `clickhouse` configuration details.

To accomodate a clean illustration for clickhouse due to the nuances, we separated out the example into the `clickhouse` directory.

The configuration for the OpenTelemetry collector should look familiar if you have followed the above examples, and this [yaml](./clickhouse/otel-collector-config.yaml) could serve as a base to fit your configuration needs.

#### Analytics

> Once you run the `docker-compose`, you can start creating flags, making evaluations etc. like regular. As data gets collected on the clickhouse server you can connect to the server via web UI exposed on the URL `http://localhost:8123/play`, and from this you can glean powerful analytical data such as average/quantile measurements from the flag evaluations.

Reviewer: ```suggestion
Once you run the `docker-compose`, you can start creating flags, making evaluations, etc. After data is collected in the Clickhouse server, you can connect to the server via web UI at `http://localhost:8123/play`. From here you can glean powerful analytical data such as average and quantile measurements from the flag evaluations.
```

!['Clickhouse Example'](../../images/clickhouse.png)

```sql
SELECT SpanName, avg(Duration), quantile(0.9)(Duration) AS p90, quantile(0.95)(Duration) AS p95, quantile(0.99)(Duration) AS p99 FROM otel.otel_traces WHERE SpanName='flipt.evaluation.EvaluationService/Variant' OR SpanName='flipt.evaluation.EvaluationService/Boolean' GROUP BY SpanName;
```

> This query will return average, and quantile results (p90, p95, p99) for `Variant` and `Boolean` evaluations. To do more cool and insighful things via their SQL syntax, you can refer to the [Clickhouse SQL reference docs](https://clickhouse.com/docs/en/sql-reference).

Reviewer: ```suggestion
This query will return average and quantile results (p90, p95, p99) for `Variant` and `Boolean` evaluations. To perform more insightful queries, see the [Clickhouse SQL reference documentation](https://clickhouse.com/docs/en/sql-reference).
```

1. Open the Datadog traces UI under the menu item on the left `APM` then `Traces`
1. You should see a list of traces to explore

### Clickhouse

#### Requirements

> 1. [Clickhouse CLI](https://clickhouse.com/docs/en/install)

Reviewer 2: You also could use `docker compose exec clickhouse clickhouse client` without installing the client on host machine.

Author: @erka Indeed. Yeah that would be better for ppl who do not want it on their machine.

Author: I think i'll remove that part, because on port `8123` you get a UI which you can query from anyway.
